include/unifex/async_scope.hpp

```diff
@@ -561,6 +564,201 @@ struct future final {
   };
 };
 
+template <typename Operation, typename Receiver>
+struct _cleaning_receiver final {
+  struct type;
+};
+
+template <typename Operation, typename Receiver>
+using cleaning_receiver =
+    typename _cleaning_receiver<Operation, Receiver>::type;
+
+template <typename Operation, typename Receiver>
+struct _cleaning_receiver<Operation, Receiver>::type final {
+  template <typename... Values>
+  void set_value(Values&&... values) noexcept {
+    op_->cleanup();
+    unifex::set_value(std::move(receiver_), (Values &&) values...);
+  }
+
+  template <typename E>
+  void set_error(E&& e) noexcept {
+    op_->cleanup();
+    unifex::set_error(std::move(receiver_), (E &&) e);
+  }
+
+  void set_done() noexcept {
+    op_->cleanup();
+    unifex::set_done(std::move(receiver_));
+  }
+
+  template(typename CPO)                       //
+      (requires is_receiver_query_cpo_v<CPO>)  //
+      friend auto tag_invoke(CPO cpo, const type& r) noexcept(
+          noexcept(static_cast<CPO&&>(cpo)(r.receiver_)))
```

Comment:
I wonder if there's a `is_nothrow_callable_v` somewhere.

Reply:
There is, fixed!